pkg/operator/endpoints/metrics.go

```diff
@@ -0,0 +1,449 @@
+		if metricData.Values == nil {
+			continue
+		}
+
+		switch {
+		case *metricData.Label == "2XX":
+			networkStats.Code2XX = *SumInt(metricData.Values...)
+		case *metricData.Label == "4XX":
+			networkStats.Code4XX = *SumInt(metricData.Values...)
+		case *metricData.Label == "5XX":
+			networkStats.Code5XX = *SumInt(metricData.Values...)
+		case *metricData.Label == "Latency":
+			latencyAvg = metricData.Values
+		case *metricData.Label == "RequestCount":
+			requestCount = metricData.Values
+		}
+	}
+
+	avg, err := Avg(latencyAvg, requestCount)
+	if err != nil {
+		return err
+	}
+	networkStats.Latency = avg
+	apiMetrics.NetworkStats = &networkStats
+	return nil
+}
+
+func ExtractClassificationMetrics(metricsDataResults []*cloudwatch.MetricDataResult, apiMetrics *schema.APIMetrics) {
+	apiMetrics.ClassDistribution = map[string]int{}
+	for _, metricData := range metricsDataResults {
+		if metricData.Values == nil {
+			continue
+		}
+
+		if strings.HasPrefix(*metricData.Label, "class_") {
+			className := (*metricData.Label)[6:]
+			apiMetrics.ClassDistribution[className] = *SumInt(metricData.Values...)
+		}
+	}
+}
+
+func ExtractRegressionMetrics(metricsDataResults []*cloudwatch.MetricDataResult, apiMetrics *schema.APIMetrics) error {
+	var regressionStats schema.RegressionStats
+	var predictionAvg []*float64
+	var requestCount []*float64
+
+	for _, metricData := range metricsDataResults {
+		if metricData.Values == nil {
+			continue
+		}
+
+		switch {
+		case *metricData.Label == "Min":
+			min := Min(metricData.Values...)
+			if min != nil {
+				regressionStats.Min = *min
+			}
+		case *metricData.Label == "Max":
+			max := Max(metricData.Values...)
+			if max != nil {
+				regressionStats.Max = *max
+			}
+		case *metricData.Label == "RequestCount":
+			count := SumInt(metricData.Values...)
+			if count != nil {
+				regressionStats.SampleCount = *count
+			}
+			requestCount = metricData.Values
+		case *metricData.Label == "Avg":
+			predictionAvg = metricData.Values
+		}
+	}
+
+	avg, err := Avg(predictionAvg, requestCount)
+	if err != nil {
+		return err
+	}
+	if avg != nil {
+		regressionStats.Avg = *avg
+	}
+	if regressionStats.SampleCount > 0 {
+		apiMetrics.RegressionStats = &regressionStats
+	}
+	return nil
+}
+
+func GetAPIDimensions(appName string, api *context.API) []*cloudwatch.Dimension {
+	return []*cloudwatch.Dimension{
+		{
+			Name:  aws.String("AppName"),
+			Value: aws.String(appName),
+		},
+		{
+			Name:  aws.String("APIName"),
+			Value: aws.String(api.Name),
+		},
+		{
+			Name:  aws.String("APIID"),
+			Value: aws.String(api.ID),
+		},
+	}
+}
+
+func GetLatencyMetricsDef(routeName string, period int64) []*cloudwatch.MetricDataQuery {
+	networkDataQueries := []*cloudwatch.MetricDataQuery{
+		{
+			Id:    aws.String("latency"),
+			Label: aws.String("Latency"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
```

Comment:
I think cluster name might be better here? We might remove namespace at some point since we don't install within existing clusters anymore.

Or maybe we can remove Namespace? Do we also scope on API name?

Also I might be totally misunderstanding this, we can discuss